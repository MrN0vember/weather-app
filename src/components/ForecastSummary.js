import React from "react";
import PropTypes from "prop-types";

const ForecastSummary = (props) => {
    const {
        date,
        description,
        icon,
        tempreature
    } = props;
    
  return <div class="forecast-summary">
      <div className="forecast-summary_date">
          {date}
          </div>
          <div className="forecast-summary_icon">
              {icon}
          </div>
          <div className="forecast-summary_tempreature">
              {tempreature.max}
              &deg;C
          </div>
          <div className="forecast-summary__description">
        {description}
      </div>
  </div>;
};

ForecastSummary.PropTypes = {
  date: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  tempreature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
};

export default ForecastSummary;
