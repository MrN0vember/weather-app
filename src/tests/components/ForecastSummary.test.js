import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
    const validProps ={
        date: 1111111,
        description: "Stub Description",
        icon: "stubIcon",
        temperature: {
            max: 14,
            min: 7,
        },
    };
    it("renders correctly", () => {
        const { asFragment } = render(
          <ForecastSummary
            date={validProps.date}
            description={validProps.description}
            icon={validProps.icon}
            temperature={validProps.temperature}
          />
        );
        expect(asFragment()).toMatchSnapshot();
      });
      it("renders the correct vaules", () => {
        const { getByText, getByTestId } = render(
          <ForecastSummary
          date={validProps.date}
          description={validProps.description}
          icon={validProps.icon}
          temperature={validProps.temperature}
        />
      );
          expect(getByText("1111111")).toHaveClass("forecast-summary_date");
          expect(getByText("Stub Description")).toHaveClass("forecast-summary_description");
          expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary_icon");
          expect(getByText("14Â°C")).toHaveClass("forecast-summary_temperature");
      });
    });
