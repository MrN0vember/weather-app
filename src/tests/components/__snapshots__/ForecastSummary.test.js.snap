// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="forecast-summary_date"
    >
      1111111
    </div>
    <div
      class="forecast-summary_icon"
      data-testid="forecast-icon"
    >
      &gt;stubIcon
    </div>
    <div
      class="forecast-summary_temperature"
    >
      14Â°C
    </div>
    <div
      class="forecast-summary_description"
    >
      Stub Description
    </div>
  </div>
</DocumentFragment>
`;
